scene/resources/atlas_texture.cpp

```diff
@@ -248,6 +248,9 @@ Ref<Image> AtlasTexture::get_image() const {
 	if (!atlas.is_valid() || !atlas->get_image().is_valid()) {
 		return Ref<Image>();
 	}
+	if (region.size.x <= 0 || region.size.y <= 0) {
+		return Ref<Image>();
+	}
 
 	return atlas->get_image()->get_region(region);
```

Comment:
I agree with the suggestion and reasoning for it, made the changes.

For the second half, I don't know how it supposed to work, but I think that for the purposes of the region editor we don't want to have the margin applied. I think it's supposed to be added after the region. But yeah, it's not new to this PR, so I'd leave it for another day if someone has a better idea.

Reply:
> For the second half, I don't know how it supposed to work, but I think that for the purposes of the region editor we don't want to have the margin applied. I think it's supposed to be added after the region. But yeah, it's not new to this PR, so I'd leave it for another day if someone has a better idea.

Yeah, I agree. Margin shouldn't be shown in the region selector, it's indeed "added after region". What would make sense for the margin is e.g. to have a separate "margin editor" dialog (openable by "Edit Margin" button in the inspector) where the already selected region would be shown and it would allow to only drag the margins on top of the region.
But yeah, that's for a different PR.